createrepo_c: 0.11.1 -> 0.17.1

maisem opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
createrepo_c doesn't build on darwin, this fixes that by updating the version to `0.17.1`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-rmcgibbo commented:
Result of `nixpkgs-review pr 120906` at a75cb32f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>createrepo_c</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/package-management/createrepo_c/default.nix:33:5:
```
   |
33 |     license     = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

rpm is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/package-management/createrepo_c/default.nix:28:3:
```
   |
28 |   buildInputs = [ bzip2 expat glib curl libxml2 python3 rpm openssl sqlite file xz pcre bash-completion zstd zchunk libmodulemd ];
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/package-management/createrepo_c/default.nix:15:5:
```
   |
15 |     ./fix-bash-completion-path.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/package-management/createrepo_c/default.nix:16:5:
```
   |
16 |     ./fix-python-install-path.patch
   |     ^
```
</li>
  </ul>
</details>

fufexan commented:
Result of `nixpkgs-review pr 120906` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>createrepo_c</li>
  </ul>
</details>

maisem commented:
/marvin opt-in
/status needs_reviewer

fufexan reviewed (changes requested):
Try to address @r-rmcgibbo's suggestions.

maisem commented:
Result of `nixpkgs-review pr 120906` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>createrepo_c</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/tools/package-management/createrepo_c/default.nix:
```diff
@@ -1,18 +1,20 @@
-{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, bzip2, expat, glib, curl, libxml2, python3, rpm, openssl, sqlite, file, xz, pcre, bash-completion }:
+{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, bzip2, expat, glib, curl, libxml2, python3, rpm, openssl, sqlite, file, xz, pcre, bash-completion, zstd, zchunk, libmodulemd }:
```
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, pkg-config, bzip2, expat, glib, curl, libxml2, python3, rpm
, openssl, sqlite, file, xz, pcre, bash-completion, zstd, zchunk, libmodulemd
}:
```

that line got a bit long.

SuperSandro2000 commented on pkgs/tools/package-management/createrepo_c/default.nix:
```diff
@@ -23,15 +25,15 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
     description = "C implementation of createrepo";
     homepage    = "http://rpm-software-management.github.io/createrepo_c/";
```
```suggestion
    homepage    = "https://rpm-software-management.github.io/createrepo_c/";
```

SuperSandro2000 commented on pkgs/tools/package-management/createrepo_c/fix-bash-completion-path.patch:
```diff
@@ -1,11 +1,11 @@
++        SET(BASHCOMP_DIR "@BASHCOMP_DIR@")
+         message("Bash completion directory: ${BASHCOMP_DIR}")
+         INSTALL(FILES createrepo_c.bash DESTINATION ${BASHCOMP_DIR} RENAME createrepo_c)
+         INSTALL(CODE "
```
Please add the final new line back.

SuperSandro2000 commented on pkgs/tools/package-management/createrepo_c/fix-python-install-path.patch:
```diff
@@ -1,11 +1,11 @@
- 
- MESSAGE(STATUS "Python install dir is ${PYTHON_INSTALL_DIR}")
+
+ MESSAGE(STATUS "Python install dir is ${PYTHON_INSTALL_DIR}")
```
Please add the final new line back.

jonringer reviewed (commented):
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

 `git rebase -i` is a powerful command which achieves this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

otherwise LGTM :)

Commit:
createrepo_c: 0.11.1 -> 0.17.1

maisem commented:
> Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.
> 
> `git rebase -i` is a powerful command which achieves this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)
> 

Rebased and squashed.

> otherwise LGTM :)

I just like being able to track changes as they come up and squashing at the end. :)

maisem commented:
@jonringer @SuperSandro2000 can either of you help merge this?

jonringer reviewed (approved):
LGTM

Result of nixpkgs-review pr 120906 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>createrepo_c</li>  </ul></details>
